lib/calendar.ts

```diff
@@ -1042,7 +1096,7 @@ const helperHebrew = ObjectAssign({}, nonIsoHelperBase, {
     } else {
       // When called without input coming from legacy Date output, simply ensure
       // that all fields are present.
-      this.validateCalendarDate(calendarDate);
+      (this as unknown as NonIsoHelperBase).validateCalendarDate(calendarDate);
```

Comment:
The error is 

```
Conversion of type <omitted for brevity> to type 'NonIsoHelperBase' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
```

I believe they don't overlap because the `this` in that context is being inferred as the collection of properties only within that object literal (and doesn't include all the properties that are present due to the `nonIsoHelperBase` within the Object.assign call). This seems to be a limitation of TypeScript's type inference, but I think it could be resolved by rewriting the calendars using classes (where this sort of extension is better understood in the type system).

Reply:
I swapped this over to specifying the `this` type as part of the function's parameters to avoid the dangerous casts. Resolving.